shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -505,6 +505,9 @@ @implementation FlutterTextInputView {
   const char* _selectionAffinity;
   FlutterTextRange* _selectedTextRange;
   CGRect _cachedFirstRect;
+  // The view has reached end of life, and is no longer
+  // allowed to access its textInputDelegate.
+  BOOL _decommissioned;
```

Comment:
Why do we need this extra state, wouldn't removal of the FlutterTextInputView be enough to say it's decommissioned?

Reply:
It seems like the crux of your problem was the reuse of the views, right?  Which you removed elsewhere.

Reply:
The crux of the problem is UIKit actually holds onto "decommissioned" views (e.g. `UITextInteraction` may keep a view alive after it's removed from its superview) and attempts to change the selection in the view (as a result the view will try to access the engine and send the update to the framework).

Reply:
I see, do we know who is holding onto the view?  It might just be a matter of clearing out it's state as well.  `_textInterfaction.view = nil`?

Reply:
(Sorry had to context switch). I think I tried that but didn't work. `UITextInteraction.textInput` is a weak ref: https://developer.apple.com/documentation/uikit/uitextinteraction/3255084-textinput?language=objc,  UIKit seems to be retaining it somewhere else and removing the weak ref doesn't seem to do anything other than setting it to `nil`.

Reply:
I'm not sure there's a public API that allows us to release the reference. I think technically it's reasonable for uikit to retain a strong ref to a `UITextInput` instance until it's not needed.

Reply:
regarding setting `_textInputDelegate` instead (the first comment), I thought about that but slightly prefer having an additional lifecycle tracking variable that's irreversible. Then I'll be able to add asserts to verify that decommissioned views are not gonna get reused. But no strong preference.

Reply:
Okay, it's kind of weird but sounds like you've looking into all the alternatives so I can go along with this.  Thanks for looking into it.